update opt-in docs (#329)
scale-it/algo-builder: docs/guide/deployer.md

#### OptIn to ASA

> For opting in to ASA, `deployer` supports following methods :-

Reviewer: ```suggestion
For opting in to ASA, `deployer` supports following methods:
```

For opting in to ASA, `deployer` supports two methods :-
- `optInAcountToASA` to opt-in to a single account signed by secret key of sender.
- `optInLsigToASA` to opt-in to a contract account (say escrow) where the account is represented by the logic signature address (`lsig.address()`).

>     - To opt in to ASA you can use either `Asset Index` or `name of the ASA`.
>     - Using Asset Index is useful when asset is not deployed using deployer.

Reviewer: ```suggestion
    To opt in to ASA you can use either `Asset Index` or `name of the ASA`. Using Asset Index is useful when asset is not deployed using deployer.
```

- There is one more method which you can use to opt-in, It can be used with group transactions also
- `executeTransaction` to opt-in single account or contract account to ASA.
- Ex: To opt-in a single account, Params will look like this:
```js
   const execParam: ExecParams = {
    type: TransactionType.OptInASA,
    sign: SignType.SecretKey,
    fromAccount: user.account,
    assetID: assetID,
    payFlags: payFlags
  };
```
- Ex: To opt-in to a contract account
```js
   const execParam: ExecParams = {
    type: TransactionType.OptInASA,
    sign: SignType.LogicSignature,
    fromAccountAddr: lsig.address(),
    assetID: assetID,
    payFlags: payFlags
    lsig: lsig
  };

> ```

Reviewer: all things below "There is one more..." should be in the same bulletpoint.

Author: done

You can learn more about the flags from [Deployer API](https://scale-it.github.io/algo-builder/);
You can learn more about Stateful Smart Contracts [here](https://developer.algorand.org/docs/features/asc1/stateful/).

#### OptIn to SSC

> For opting in to SSC, `deployer` supports the following methods :-

Reviewer: ```suggestion
For opting in to SSC, `deployer` supports the following methods:
```

You can learn more about the flags from [Deployer API](https://scale-it.github.io/algo-builder/);
You can learn more about Stateful Smart Contracts [here](https://developer.algorand.org/docs/features/asc1/stateful/).

#### OptIn to SSC

> For opting in to SSC, `deployer` supports two methods :-

Reviewer: ```suggestion
For opting in to SSC, `deployer` supports the following methods :-
```

- `optInAcountToSSC` to opt-in to a single account signed by secret key of sender.
- `optInLsigToSSC` to opt-in to a contract account (say escrow) where the account is represented by the logic signature address (`lsig.address()`).

>   - To opt in to SSC you can use `Application Index`.

Reviewer: can you add a link or explain it further?

Author: yes

- There is one more method which you can use to opt-in, It can be used with group transactions also
- `executeTransaction` to opt-in single account or contract account to SSC.
- Ex: To opt-in a single account, Params will look like this:
```js
   const execParam: ExecParams = {
    type: TransactionType.OptInSSC,
    sign: SignType.SecretKey,
    fromAccount: user.account,
    appID: appID,
    payFlags: payFlags
  };
```
- Ex: To opt-in to a contract account
```js
   const execParam: ExecParams = {
    type: TransactionType.OptInSSC,
    sign: SignType.LogicSignature,
    fromAccountAddr: lsig.address(),
    appID: appID,
    payFlags: payFlags
    lsig: lsig
  };

> ```

Reviewer: same as above. Maybe we can reword it, a bit? 

"Like with ASA, we can also use `executeTransaction` to opt-in a single account or contract account to SSC. Example: ..."

Author: ok
